### PR TITLE
feat(extract): add CLI flags for extraction modes and enrichment orchestrator

### LINKED ISSUE #166: [M2-D2.6] Add CLI flags for extraction modes
### Context
**PRD:** `docs/project/PRD/active/PRD-phase-11-metadata-extraction.md` — Milestone 2, Deliverable D2.6

The extraction CLI needs flags to control strictness (fail vs warn on missing fields), enable two-pass workflows (components-only then enrich), and manage output.

### Traceability
- **PRD Section:** M2-D2.6 (CLI flags and modes)
- **Success Criteria:** SC-5 (riviere extract --pr command available)

### Acceptance Criteria
- [ ] `--allow-incomplete` flag: warn and continue, output includes `_missing` arrays
- [ ] `--components-only` flag: skip metadata extraction (Phase 10 behavior)
- [ ] `--enrich <file>` flag: read draft components from file, apply extraction rules
- [ ] `--dry-run` flag: validate config, report what would be extracted, no output file
- [ ] `-o <file>` flag: output to file instead of stdout
- [ ] Exit codes implemented: 0 (success), 1 (extraction failure), 2 (config validation), 3 (runtime error)
- [ ] Error messages follow format: `Error: [CATEGORY] CODE\n  Description.\n  Location: file:line\n  Suggestion: how to fix`
- [ ] Categories: CONFIG_VALIDATION, EXTRACTION_FAILURE, GIT_ERROR, RUNTIME_ERROR
- [ ] All flags documented in --help output

### Edge Case Scenarios
- `--allow-incomplete` with all fields present → no `_missing` array, exit 0
- `--allow-incomplete` with missing fields → `_missing` array populated, warnings printed, exit 0
- Default (strict) with missing fields → exit 1 with error listing missing fields
- `--enrich` with invalid JSON → exit 3 (runtime error)
- `--enrich` with file not found → exit 3 (runtime error)
- `--dry-run` with invalid config → exit 2 (config validation)
- `-o` to read-only path → exit 3 (runtime error)
- Combined `--components-only --enrich` → error (mutually exclusive)

### Implementation Guidelines
- **Location:** `packages/riviere-cli/src/commands/extract/extract.ts`
- Use Commander.js (already in use for CLI)
- Follow existing flag patterns in CLI
- Lenient mode adds `_missing: string[]` to component output
- Output format matches PRD Section 3.2 (version, extractedAt, config, components, warnings)
- Error formatting should be consistent across all error types

### Embedded Reasoning
**Why:** Different workflows need different strictness. CI wants strict (fail fast). Incremental adoption wants lenient. Two-pass enables AI-assisted completion.
**What:** 5 new flags controlling extraction behavior and output.
**How:** Commander.js options, conditional logic in extract command, structured error handling.

### Testing Strategy
- **Unit:** Flag parsing and validation
- **Integration:** CLI invocation with each flag combination
- **Edge Cases:** All scenarios from Edge Case Scenarios section
- **Error Handling:** Verify exit codes and error message format

### Dependencies
Depends on #165

### Verification
```bash
pnpm nx build riviere-cli
# Check help output
./packages/riviere-cli/dist/bin.js extract --help | grep -E '(allow-incomplete|components-only|enrich|dry-run)'
```

---

## Amendment

Amendment: The acceptance criteria for error message format and categories should reflect the existing JSON output pattern used by the CLI (formatError/formatSuccess in output.ts). The codebase uses structured JSON output ({success, error: {code, message, suggestions}}) rather than human-readable text. The CliErrorCode enum already defines error categories (ConfigNotFound, ValidationError). The --help output is auto-generated by Commander.js from option definitions and requires no separate implementation.

### PR DESCRIPTION
**Pull request by NTCoding**
Closes #166

### Context
**PRD:** `docs/project/PRD/active/PRD-phase-11-metadata-extraction.md` — Milestone 2, Deliverable D2.6

The extraction CLI needs flags to control strictness (fail vs warn on missing fields), enable two-pass workflows (components-only then enrich), and manage output.

### Traceability
- **PRD Section:** M2-D2.6 (CLI flags and modes)
- **Success Criteria:** SC-5 (riviere extract --pr command available)

### Acceptance Criteria
- [ ] `--allow-incomplete` flag: warn and continue, output includes `_missing` arrays
- [ ] `--components-only` flag: skip metadata extraction (Phase 10 behavior)
- [ ] `--enrich <file>` flag: read draft components from file, apply extraction rules
- [ ] `--dry-run` flag: validate config, report what would be extracted, no output file
- [ ] `-o <file>` flag: output to file instead of stdout
- [ ] Exit codes implemented: 0 (success), 1 (extraction failure), 2 (config validation), 3 (runtime error)
- [ ] Error messages follow format: `Error: [CATEGORY] CODE\n  Description.\n  Location: file:line\n  Suggestion: how to fix`
- [ ] Categories: CONFIG_VALIDATION, EXTRACTION_FAILURE, GIT_ERROR, RUNTIME_ERROR
- [ ] All flags documented in --help output

### Edge Case Scenarios
- `--allow-incomplete` with all fields present → no `_missing` array, exit 0
- `--allow-incomplete` with missing fields → `_missing` array populated, warnings printed, exit 0
- Default (strict) with missing fields → exit 1 with error listing missing fields
- `--enrich` with invalid JSON → exit 3 (runtime error)
- `--enrich` with file not found → exit 3 (runtime error)
- `--dry-run` with invalid config → exit 2 (config validation)
- `-o` to read-only path → exit 3 (runtime error)
- Combined `--components-only --enrich` → error (mutually exclusive)

### Implementation Guidelines
- **Location:** `packages/riviere-cli/src/commands/extract/extract.ts`
- Use Commander.js (already in use for CLI)
- Follow existing flag patterns in CLI
- Lenient mode adds `_missing: string[]` to component output
- Output format matches PRD Section 3.2 (version, extractedAt, config, components, warnings)
- Error formatting should be consistent across all error types

### Embedded Reasoning
**Why:** Different workflows need different strictness. CI wants strict (fail fast). Incremental adoption wants lenient. Two-pass enables AI-assisted completion.
**What:** 5 new flags controlling extraction behavior and output.
**How:** Commander.js options, conditional logic in extract command, structured error handling.

### Testing Strategy
- **Unit:** Flag parsing and validation
- **Integration:** CLI invocation with each flag combination
- **Edge Cases:** All scenarios from Edge Case Scenarios section
- **Error Handling:** Verify exit codes and error message format

### Dependencies
Depends on #165

### Verification
```bash
pnpm nx build riviere-cli
# Check help output
./packages/riviere-cli/dist/bin.js extract --help | grep -E '(allow-incomplete|components-only|enrich|dry-run)'
```

---

## Amendment

Amendment: The acceptance criteria for error message format and categories should reflect the existing JSON output pattern used by the CLI (formatError/formatSuccess in output.ts). The codebase uses structured JSON output ({success, error: {code, message, suggestions}}) rather than human-readable text. The CliErrorCode enum already defines error categories (ConfigNotFound, ValidationError). The --help output is auto-generated by Commander.js from option definitions and requires no separate implementation.

## Summary by CodeRabbit

* **New Features**
  * File output for extraction (-o/--output), components-only view, draft enrichment (--enrich), and allow-partial-success (--allow-incomplete)
  * Programmatic enrichment API exposed
  * Workflow timing capture and optional timings.md output

* **Bug Fixes**
  * Standardized CLI exit codes and structured JSON error reporting
  * Improved handling of unwritable output paths and CI polling behavior

* **Tests**
  * Expanded suites for extraction, enrichment, draft loading, file I/O, timing, and edge cases

* **Documentation**
  * Reorganized conventions and workflow docs with standardized rule identifiers

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: .claude/agents/code-review.md [hunk 1]
```diff
@@ -7,58 +7,95 @@ color: purple
...
+- **N/A**: Rule doesn't apply to changed files. State why.
+
+Rule sets to audit (every ID must appear):
+- Software Design: SD-001 through SD-021
+- Anti-Patterns: AP-001 through AP-006
+- Standard Patterns: SP-001, SP-002
+- Testing: TS-001 through TS-011
+- Codebase Structure: CS-001 through CS-008
+- Code Review: CR-001 through CR-006
+
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply .claude/agents/code-review.md]: ❌ **Rejected**: The exhaustive audit is the entire point of this change. The previous reviewer used category-level checkboxes and missed SD-019 violations while praising the violating code. N/A rows prove the rule was considered — without them, skipped rules are indistinguishable from forgotten rules. The overhead is borne by the agent, not the developer. Scoped audits would reintroduce the exact blind spots this change was designed to eliminate.

---
*Generated by Claude Code*

File: docs/conventions/software-design.md [hunk 2]
```diff
@@ -4,19 +4,19 @@ Professional software design patterns and principles for writing maintainable, w
 
 ## Critical Rules
 
-🚨 **Fail-fast over silent fallbacks.** Never use fallback chains (`value ?? backup ?? 'unknown'`). If data should exist, validate and throw a clear error.
+🚨 **SD-001: Fail-fast over silent fallbacks.** Never use fallback chains (`value ?? backup ?? 'unknown'`). If data should exist, validate and throw a clear error.
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply docs/conventions/software-design.md]: ❌ **Rejected**: There is only one ## heading with SD-001 (line 132). Line 7 is inline bold text in the Critical Rules summary section, not a heading. No duplicate anchor exists. Grep confirms: only one match for '^## SD-001'.

---
*Generated by Claude Code*

File: docs/conventions/software-design.md [hunk 3]
```diff
@@ -362,7 +362,7 @@ const processedOrder = processOrder(myOrder)
 - Prefer `map`/`filter`/`reduce` over `forEach` with mutation
 - If you must mutate, make it explicit and contained
 
-## YAGNI - You Aren't Gonna Need It
+## SD-021: YAGNI - You Aren't Gonna Need It
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply docs/conventions/software-design.md]: ❌ **Rejected**: YAGNI stands for 'You Aren't Gonna Need It' — this is the canonical expansion coined by Ron Jeffries in Extreme Programming. Changing it to 'going to' would misquote the established term. The heading preserves the original wording intentionally.

---
*Generated by Claude Code*
